pkg/skaffold/jib/jib.go

```diff
@@ -17,64 +17,173 @@ limitations under the License.
 package jib
 
 import (
+	"encoding/json"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"sort"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/karrick/godirwalk"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-func getDependencies(cmd *exec.Cmd) ([]string, error) {
-	stdout, err := util.RunCmdOut(cmd)
-	if err != nil {
-		return nil, err
+// pathMap contains fields parsed from Jib's JSON output
+type pathMap struct {
+	// Build lists paths to build definitions that trigger a call out to Jib to refresh the pathMap, as well as a rebuild, upon changing
+	Build []string `json:"build"`
+
+	// Inputs lists paths to build dependencies that trigger a rebuild upon changing
+	Inputs []string `json:"inputs"`
+
+	// Results lists paths to files that should be ignored when checking for changes to rebuild
+	Results []string `json:"ignore"`
+}
+
+// filesLists contains cached build/input dependencies
+type filesLists struct {
+	// PathMap saves the most recent output of the Jib Skaffold files task/goal
+	PathMap pathMap
```

Comment:
I might make sense to just "embed" this field. See https://travix.io/type-embedding-in-go-ba40dd4264df. Not sure